### PR TITLE
Regex pattern improvements.

### PR DESCRIPTION
**Pull request by siroky**
- Strongly typed patterns.
- Introduced identifier for UUID.
- Removed SafeMatches which was unnecessary and faulty for whitespace/empty patterns.
- Added xunit test runner for Visual Studio.

### CODE REVIEW

File: src/Mews.Eet/Communication/SendRevenueMessage.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ private RevenueHeader GetRevenueHeader()
         {
             return new RevenueHeader
             {
-                MessageUuid = RevenueRecord.Identifier.ToString(),
+                MessageUuid = RevenueRecord.Identifier.Value.ToString(),
```
Comment (jirihelmich) [inline src/Mews.Eet/Communication/SendRevenueMessage.cs]: `RevenueRecord.Identifier.ToString()` should remain here. `Identifier` has `ToString()` implemented exactly as you suggests. And you want to "render" the identifier, you do not really care about its inner workings. Or what was your motivation?

File: src/Mews.Eet/Dto/Identifiers/Uuid4.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Mews.Eet.Dto.Identifiers
+{
+    /// <summary>
+    /// The UUID used by EET should comply with UUID v4.0.
+    /// </summary>
+    public sealed class Uuid4 : StringIdentifier
```
Comment (jirihelmich) [inline src/Mews.Eet/Dto/Identifiers/Uuid4.cs]: I would create `GuidIdentifier` class derived from Identifier, I'm not sure this inheritance is correct compared to the rest of the identifiers.

File: src/Mews.Eet/Communication/SendRevenueMessage.cs (hunk 1 above)
Comment (siroky) [inline src/Mews.Eet/Communication/SendRevenueMessage.cs]: I wanted to make sure that change in implementation of ToString (to e.g. `<UUID("....")>` or whatever else that could be suitable for debug purposes) would not break the communication.

File: src/Mews.Eet/Dto/Identifiers/Uuid4.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    {
+        public Uuid4(Guid guid)
+            : this(guid.ToString())
+        {
+        }
+
+        public Uuid4(string value)
+            : base(value, Patterns.Uuid4)
```
Comment (jirihelmich) [inline src/Mews.Eet/Dto/Identifiers/Uuid4.cs]: Having the validation is good, even when we know that this is used just internally from the library. Thanks for this. Now I see the motivation for inheriting this from `StringIdentifier`, I'm just still not sure it's correct.

**Comment by jirihelmich** [comment]
Regex update.

**Pull request merged**